Source/JavaScriptCore/runtime/LiteralParser.cpp

```diff
@@ -1263,15 +1265,28 @@ JSValue LiteralParser<CharType, reviverMode>::parseRecursivelyEntry(VM& vm)
         return parse(vm, StartParseExpression, nullptr);
     TokenType type = m_lexer.currentToken()->type;
     if (type == TokLBrace || type == TokLBracket)
-        return parseRecursively(vm, std::bit_cast<uint8_t*>(vm.softStackLimit()));
+        return parseRecursively<ParserMode::StrictJSON>(vm, std::bit_cast<uint8_t*>(vm.softStackLimit()));
     return parsePrimitiveValue(vm);
 }
 
 template<typename CharType, JSONReviverMode reviverMode>
+JSValue LiteralParser<CharType, reviverMode>::evalRecursivelyEntry(VM& vm)
+{
+    ASSERT(m_mode == SloppyJSON);
```

Comment:
Since this is only called from `tryEval`, please also add a `static_assert(reviverMode == JSONReviverMode::Disabled));` here to document this invariant.

Reply:
Ditto.